web3rpc/refactor-code-example.js

```diff
@@ -1,84 +1,125 @@
+  fs.readdir(path, async (err, files) => {
+    files.forEach((file) => {
+      if (file.endsWith('api.mdx')) {
+        filesToRefactor.push(path + '/' + file)
+      }
     })
+    const promises = filesToRefactor.map((filePath) =>
+      refactorOpenAPI(filePath)
+    )
+    await Promise.all(promises)
+  })
 }
 
-const refactorOpenAPI = async ( path ) => {
-    let fileContents = fs.readFileSync(path, 'utf8')
-    let apiLine = '';
-    let apiIndex;
-    
-    const lines = fileContents.split("\n");
-    lines.forEach((line, index) => {
-        if (line.startsWith("api: ")) {
-            apiLine = line.split("api: ")[1];
-            apiIndex = index;
-        }
-    });
+const refactorOpenAPI = async (path) => {
+  let fileContents = fs.readFileSync(path, 'utf8')
+  let apiLine = ''
+  let apiIndex
+
+  const lines = fileContents.split('\n')
+  lines.forEach((line, index) => {
+    if (line.startsWith('api: ')) {
+      apiLine = line.split('api: ')[1]
+      apiIndex = index
+    }
+  })
+
+  const decompressedOpenAPI = await flateOpenAPI(apiLine)
+
+  // remove the redundant postman url path
+  // because the RPC APIs of kaiachain does not have the path
+  decompressedOpenAPI.postman.url.path = []
+  const compressedOpenAPI = await deflateOpenAPI(
+    JSON.stringify(decompressedOpenAPI)
+  )
+
+  lines[apiIndex] = `api: ${compressedOpenAPI}`
 
-    const decompressedOpenAPI = await flateOpenAPI(apiLine);
-    
-    // remove the redundant postman url path
-    // because the RPC APIs of kaiachain does not have the path
-    decompressedOpenAPI.postman.url.path = [];
-    const compressedOpenAPI = await deflateOpenAPI(
-        JSON.stringify(decompressedOpenAPI)
-    );
-    
-    lines[apiIndex] = `api: ${compressedOpenAPI}`;
+  let newFileContents = lines.join('\n')
 
-    let newFileContents = lines.join('\n');
+  // remove unwanted params from request and response
+  newFileContents = removeSchemaItem('id', newFileContents)
+  newFileContents = removeSchemaItem('method', newFileContents)
+  newFileContents = removeSchemaItem('jsonrpc', newFileContents)
 
-     // remove unwanted params from request and response
-     newFileContents=removeSchemaItem('id',newFileContents)
-     newFileContents=removeSchemaItem('method',newFileContents)
-     newFileContents=removeSchemaItem('jsonrpc',newFileContents)
- 
-     // change <details or </details to <div or </div (details tag make description collapsed, we want it to be shown)
-     newFileContents=newFileContents.replace(/<details/gi,'<div') 
-     newFileContents=newFileContents.replace(/<\/details/gi,'</div') 
+  newFileContents = formatFoldedParams(newFileContents)
 
-    fs.writeFileSync(path, newFileContents);
+  fs.writeFileSync(path, newFileContents)
 }
-const removeSchemaItem=(name,str='')=>{
-    return str.replace(new RegExp(`<\\s*SchemaItem\\b[^>]*\\bname\\s*=\\s*{"${name}"}[^>]*>\\s*</\\s*SchemaItem\\s*>`,'g'),'')
+const formatFoldedParams = (content = '') => {
```

Comment:
`formatFoldedParams` search for error session wrapped with details tag before changing details tag to div tag then keep it unchanged.